### PR TITLE
Fix recruitment banner issues

### LINKED ISSUE #541: Recruitment banner showing up for non-ICS majors
image

No idea why this is the case. I think it didn't use to be this way.

Also, the text overflows.

### PR DESCRIPTION
**Pull request by Voark**
## Summary
Fixed bug where recruitment banner shows for any major, and also fixed text overflow issues.
The recruitment banner is now an alert that pops up on the bottom right.

image
image

## Test Plan
Check that recruitment banner only shows for right majors and that the text looks fine.

Closes #541 

## Future Followup
If we keep the alert/snackbar style implementation it might be worth integrating it with the rest of our snackbar and moving it to a separate component but I did not want to set up another global state variable.

### CODE REVIEW

File: src/components/RightPane/CoursePane/CourseRenderPane.tsx [hunk 1]
```diff
@@ -277,27 +273,30 @@ class CourseRenderPane extends PureComponent<CourseRenderPaneProps, CourseRender
...
-                            <img src={isDarkMode() ? darkNoNothing : noNothing} alt="No Results Found" />
-                        </div>
-                    ) : (
-                        this.state.courseData.map((_: School | Department | AACourse, index: number) => {
-                            let heightEstimate = 200;
-                            if ((this.state.courseData[index] as AACourse).sections !== undefined)
-                                heightEstimate =
-                                    (this.state.courseData[index] as AACourse).sections.length * 60 + 20 + 40;
+                    <div className={classes.root} style={{ position: 'relative' }}>
+                        <div className={classes.spacing} />
```
Comment (Voark) [inline src/components/RightPane/CoursePane/CourseRenderPane.tsx:282]: This div is literally just there to space the back and reload buttons so they don't overlap.
image
image

If anyone else knows a better way to do this LMK.

**Comment by Voark** [comment]
I just realized I forgot to turn off my dark mode extension so the colors aren't right, will fix it.

File: src/components/RightPane/CoursePane/CourseRenderPane.tsx [hunk 2]
```diff
@@ -102,39 +95,42 @@ const RecruitmentBanner = (classes: ClassNameMap) => {
...
+        recruitmentDismissalTime !== null &&
+        Date.now() - parseInt(recruitmentDismissalTime) < 11 * 7 * 24 * 3600 * 1000;
     const displayRecruitmentBanner =
         bannerVisibility &&
-        (window.localStorage.getItem('recruitmentDismissalTime') === null ||
-            (Date.now() - parseInt(window.localStorage.getItem('recruitmentDismissalTime') as string) >
-                11 * 7 * 24 * 3600 * 1000 &&
-                ['COMPSCI', 'IN4MATX', 'I&C SCI', 'STATS'].includes(RightPaneStore.getFormData().deptValue)));
+        !dismissedRecently &&
+        ['COMPSCI', 'IN4MATX', 'I&C SCI', 'STATS'].includes(RightPaneStore.getFormData().deptValue);
```
Comment (MinhxNguyen7) [inline src/components/RightPane/CoursePane/CourseRenderPane.tsx]: Nitpicking, but I'd extract this to be a separate variable, e.g., `is_ICS` if for nothing other than having all boolean variables.

**Review by MinhxNguyen7 — COMMENTED** [review]
Looks good code-wise, but I have reservations about using an alert because it is somewhat intrusive. @EricPedley, what's your opinion? 

Also, I don't love how light it is in dark mode, and (this is pretty minor) that blue color doesn't exist anywhere else in the UI, so we're introducing a whole new color. Maybe it should just be white/grey. I'm not sure though.

**Comment by Voark** [comment]
> Looks good code-wise, but I have reservations about using an alert because it is somewhat intrusive. @EricPedley, what's your opinion?
> 
> Also, I don't love how light it is in dark mode, and (this is pretty minor) that blue color doesn't exist anywhere else in the UI, so we're introducing a whole new color. Maybe it should just be white/grey. I'm not sure though.

Yeah will get to the colors tomorrow, didn't feel like it today. 

Figuring out sizing in that small spacing between the buttons and the edge of the table was too hard so I didn't feel like messing with it. In terms of the intrusiveness I think because the alert won't pop up again after it's closed once it's not too bad? I can also move it to bottom left if it would be less intrusive there. Otherwise someone else will have to take over for me because I cannot style a component properly to fit in that space without overflow.

File: src/components/RightPane/CoursePane/CourseRenderPane.tsx [hunk 3]
```diff
@@ -98,43 +91,49 @@ const flattenSOCObject = (SOCObject: WebsocResponse): (School | Department | AAC
...
-                            We have opportunities for experienced devs and those with zero experience!
-                        </div>
-
-                        <Button
+                <Alert
+                    icon={false}
+                    severity="info"
+                    style={{
+                        color: isDarkMode() ? '#ece6e6' : '#2e2e2e',
+                        backgroundColor: isDarkMode() ? '#2e2e2e' : '#ece6e6',
```
Comment (Voark) [inline src/components/RightPane/CoursePane/CourseRenderPane.tsx:113]: If there is a better way to theme this LMK.

Comment (EricPedley) [reply src/components/RightPane/CoursePane/CourseRenderPane.tsx:113]: I can't think of anything. Turns out the `color` prop of Alert doesn't let you enter `primary` or `secondary` like buttons and stuff, only `info`, `error`, etc.

File: src/components/RightPane/CoursePane/CourseRenderPane.tsx (hunk 1 above)
Comment (EricPedley) [reply src/components/RightPane/CoursePane/CourseRenderPane.tsx:282]: 🤷‍♂️

**Review by EricPedley — APPROVED** [review]
Thanks for taking care of this. It's been pretty ironic that the banner advertising web dev positions was badly styled. Reminds me of this: 
image
I think by being in the bottom right it isn't that intrusive, and I just want to ship *some* fix right now so it isn't broken. If someone wants to style this to be less intrusive the change would be welcome.

**Pull request merged**